server/devdocs/bin/generate-components-usage-counts

```diff
@@ -0,0 +1,174 @@
+			process.exit( 1 );
+		} );
+}
+
+/**
+ * Calculate the modules dependencies.
+ * Produces an object that has the following structure:
+ * {
+ *  'A': [dep1, dep2, dep3, ...],
+ *  'B': [dep1, dep2, dep3, ...]
+ * }
+ * @param {string} root BasePath to resolve the files' path
+ * @param {Array} fileList A list of file names
+ * @returns {Promise}
+ */
+
+function getModulesWithDependencies( root, fileList ) {
+	return new Promise( function ( resolve, reject ) {
+		var filesContent = {},
+		dependencies = {};
+
+		async.each(
+			fileList,
+			function ( fileWithPath, next ) {
+				fs.readFile(
+					fspath.join( root, fileWithPath ),
+					{ encoding: 'utf8' },
+					function ( err, data ) {
+						if ( err ) {
+							return next( err );
+						}
+						// get the dependencies
+						filesContent[ fileWithPath ] = data;
+						next( null );
+					}
+				);
+			},
+			function ( err ) {
+				if ( err ) {
+					return reject( err );
+				}
+				Object.keys( filesContent ).reduce(function ( target, module ) {
+					var deps = getDependencies( filesContent[ module ] );
+					if ( deps.length ) {
+						target[ module ] = deps;
+					}
+					return target;
+				} , dependencies );
+
+				return resolve( dependencies );
+			}
+		);
+	} );
+}
+
+/**
+ * Extract all required modules from a `code` string.
+ * Copyright (c) 2014-present, Facebook, Inc. See CREDITS.md#L1383
+ */
+function getDependencies( code ) {
+	var cache = Object.create(null);
+	var deps = [];
+
+	function addDependency( dep ) {
+		if ( !cache[ dep ] ) {
+			cache[ dep ] = true;
+			deps.push(dep);
+		}
+	}
+
+	code
+		.replace( replacePatterns.BLOCK_COMMENT_RE, '' )
+		.replace( replacePatterns.LINE_COMMENT_RE, '' )
+		.replace( replacePatterns.IMPORT_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+			return match;
+		} )
+		.replace( replacePatterns.EXPORT_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+			return match;
+		} )
+		.replace( replacePatterns.REQUIRE_RE, function ( match, pre, quot, dep, post ) {
+			addDependency( dep );
+		} );
+
+	return deps;
+}
+
+/**
+ * Given an object of modules with dependencies
+ * generate a stats object with the usage counts:
+ * {
+ *  'A': { count: 2 },
+ *  'B': { count: 10 }
+ * }
+ * @param {object} modules An object of modules with dependencies
+ * @returns {object} An object with the usage stats for each dependency
+ */
+function generateStats( modules ) {
+	return Object.keys( modules ).reduce( function ( target, moduleName ) {
+		var deps = modules[ moduleName ];
+		deps.forEach( function ( dependency ) {
+			if ( ! target[ dependency ] ) {
+				target[ dependency ] = { count: 0 };
+			}
+			target[ dependency ].count += 1;
+		} );
+		return target;
+	}, {} );
+}
+
+function writeStats( stats, statsPath ) {
+	var js = 'module.exports.stats = ' + jsFromJSON( JSON.stringify( stats, null, "\t" ) ) + ';';
```

Comment:
Do we need the `module.exports.stats` part? I just tried requiring a JSON file and it worked fine.